pkg/cmd/pulumi/pulumi.go

```diff
@@ -713,7 +715,20 @@ type cachedVersionInfo struct {
 func getUpgradeMessage(latest semver.Version, current semver.Version, isDevVersion bool) string {
 	cmd := getUpgradeCommand(isDevVersion)
 
-	msg := fmt.Sprintf("A new version of Pulumi is available. To upgrade from version '%s' to '%s', ", current, latest)
+	// If the current version is "very old", we'll return a more urgent message. "Very old" is defined as more than 24
+	// minor versions behind when the major versions are the same. Assuming a release cadence of on average 1 minor
+	// version per week, this translates to roughly 6 months. Note that we don't consider major version differences, since
+	// it's hard to know what we'd want to do in those cases. E.g. it might be that a new version of Pulumi is radically
+	// different, rather than "just improved", and so we don't want to warn about that.
+	prefix := "A new version of Pulumi is available."
+
+	minorDiff := diffMinorVersions(current, latest)
+	if current.LT(latest) && (minorDiff > 24) {
```

Comment:
Why doesn't diffMinorVersions just return a signed result making this be just (minorDiff > 24)? Not sure how current could ever be greater than latest, but it if was that would then return -24 which is less than so wouldn't trigger.

Reply:
Yeh I think a cast is safe here